File: backend/src/main/java/com/flowlink/common/error/NotFoundException.java
package com.flowlink.common.error;

/** 요청한 리소스를 찾지 못했을 때. (HTTP 404로 매핑) */
public class NotFoundException extends RuntimeException {

    public NotFoundException(String message) {
        super(message);
    }

    public static NotFoundException of(String type, Object id) {
        return new NotFoundException(type + " not found: " + id);
    }
}

File: backend/src/main/kotlin/com/flowlink/common/error/NotFoundException.kt
package com.flowlink.common.error

/** 요청한 리소스를 찾지 못했을 때. (HTTP 404로 매핑) */
class NotFoundException(message: String) : RuntimeException(message) {

    companion object {
        @JvmStatic
        fun of(type: String, id: Any?): NotFoundException =
            NotFoundException("$type not found: $id")
    }
}
